src/client/util.ts

```diff
@@ -32,12 +38,18 @@ export class Util {
         );
 
         for (let guild of guildIds) {
-            const guildCommandData = guildCommands.filter(x => x.ServerId == guild)
-                .filter(x => x.Command.CommandBuilder)
-                .flatMap(x => x.Command.CommandBuilder);
+            const guildCommandData: SlashCommandBuilder[] = [];
+
+            for (let command of guildCommands.filter(x => x.ServerId == guild)) {
+                if (!command.Command.CommandBuilder) continue;
+
+                if ((command.Environment & CoreClient.Environment) == CoreClient.Environment) {
```

Comment:
Should use `environment &= CoreClient.Environment` to be consistent with code in `client.ts`